src/tweenable.rs

```diff
@@ -894,264 +915,371 @@ mod tests {
+        )
+        .with_direction(TweeningDirection::Forward)
+        .with_repeat_count(RepeatCount::Infinite)
+        .with_repeat_strategy(RepeatStrategy::Repeat);
+
+        let expected_values = [duration * 10, Duration::MAX, Duration::MAX]
+            .into_iter()
+            .zip_eq(izip!(
+                once(10.).chain(repeat(completions as f32)),
+                [10, 3458764514, 3458764514],
+                [1, 2, 2],
+                repeat(TweeningDirection::Forward),
+                repeat(TweenState::Active),
+                once(Transform::from_translation(Vec3::splat(10.))).chain(repeat(
+                    Transform::from_translation(Vec3::splat(completions as f32)),
+                )),
+            ));
+
+        validate_tween(tween, expected_values);
+    }
+
     #[test]
-    fn tween_dir() {
+    fn tween_tick_loop_finite_large_jump() {
+        let duration = Duration::from_secs_f64(4. / 3.);
+        let tween = Tween::new(
+            EaseMethod::Linear,
+            duration,
+            TransformPositionLens {
+                start: Vec3::ZERO,
+                end: Vec3::ONE,
+            },
+        )
+        .with_direction(TweeningDirection::Forward)
+        .with_repeat_count(RepeatCount::Finite(100))
+        .with_repeat_strategy(RepeatStrategy::Repeat);
+
+        let expected_values = [duration * 10, Duration::MAX, Duration::MAX]
+            .into_iter()
+            .zip_eq(izip!(
+                [10., 100., 100.],
+                [10, 100, 100],
+                [1, 2, 2],
+                repeat(TweeningDirection::Forward),
+                once(TweenState::Active).chain(repeat(TweenState::Completed)),
+                once(Transform::from_translation(Vec3::splat(10.)))
+                    .chain(repeat(Transform::from_translation(Vec3::splat(100.)))),
+            ));
+
+        validate_tween(tween, expected_values);
+    }
+
+    #[test]
+    fn tween_tick_loop_ping_pong() {
+        let tween = Tween::new(
+            EaseMethod::Linear,
+            Duration::from_secs_f64(1. / 3.),
+            TransformPositionLens {
+                start: Vec3::ZERO,
+                end: Vec3::ONE,
+            },
+        )
+        .with_direction(TweeningDirection::Forward)
+        .with_repeat_count(RepeatCount::Infinite)
+        .with_repeat_strategy(RepeatStrategy::MirroredRepeat);
+
+        let expected_values = repeat(Duration::from_millis(200)).take(4).zip_eq(izip!(
+            successors(Some(0.6), |progress| Some(progress + 0.6)),
+            [0, 1, 1, 2],
+            [0, 1, 1, 2],
+            [
+                TweeningDirection::Forward,
+                TweeningDirection::Backward,
+                TweeningDirection::Backward,
+                TweeningDirection::Forward,
+            ],
+            repeat(TweenState::Active),
+            [
+                Transform::from_translation(Vec3::splat(0.6)),
+                Transform::from_translation(Vec3::splat(-0.2)),
```

Comment:
How is it ever possible that a tween animating a transform between x=0 and x=1 produces a negative x value?!

Reply:
Because `main` is broken. :) Progress currently increases forever so when we pong you get `1. - 1.2`.

I'm actually super happy that you spotted this because it was broken in #19 but the tests were too complicated for us to even notice.

Reply:
I'm now in favor of progress being just this thing always in `[0-1]` that only tells you how far along you are on a single loop. Anything else, especially since it's (or will be) a second-class concept, is too complicated I think and not valuable anymore.

Reply:
I'm personally still in favor of not including a progress API at all since it's not that hard to implement yourself with whatever semantics you like. That said, this won't be used internally so it doesn't matter too much and I'm fine implementing it either way.

Reply:
Though that's discussionfor #38

Reply:
Can we write the test as if there was no bug, and merge a failing test, rather than merge a test that ensure there is a bug? I'm uncomfortable having tests wrong on purpose.

Reply:
Or, comment out the test for now if the concern is CI will fail.

Reply:
I actually disagree on this one, opinions forthcoming.

Tests validate behavior, not correctness — only a human can assert correctness. In other words, tests document the current behavior.

By removing the tests, we lose the ability to see how the code currently works. That has value in and of itself: someone believing something is broken can find a test that confirms their understanding of the current behavior and then propose new behavior that the fix must adhere.

Furthermore, if there are no existing tests, when a change comes to "fix" the behavior we won't have old tests to validate the bug was actually fixed. That is, adding a new tests prevents you from seeing the transition from incorrect to correct, meaning you can't verify that the test is doing anything at all because you as the reviewer have never seen it fail. Again, this transition provides effortless documentation of the change in behavior that would otherwise be invisible... are the new tests just adding more coverage or validating some new behavior? Who knows!

Adding tests that validate behavior, right or wrong, document the state of the world and its transitions.